griptape/tasks/base_task.py

```diff
@@ -31,7 +34,7 @@ class State(Enum):
 
     @property
     @abstractmethod
-    def input(self) -> BaseArtifact:
+    def input(self) -> BaseArtifact | tuple[BaseArtifact, ...]:
```

Comment:
Descendent tasks are no longer guaranteed to return a single artifact here (see ImageVariationTask vs. ExtractionTask). Instead of widening this to `Any`, I've covered all our current bases by indicating either an artifact or a tuple of artifacts.

Reply:
when would we use tuples vs. ListArtifact? Do we eventually need a DictArtifact?

Reply:
@collindutter that's right, the task itself returns a single artifact. This was in reference to the input method specifically as its signature changes depending on the task.

Reply:
@SavagePencil tuples are used here vs. list artifacts due to the behavior of the tasks that accept multiple inputs, like, for example, ImageVariationTask. This accepts a specific number and type of artifacts as input. By instead accepting a ListArtifact we'd lose out on the explicit typing of the inputs.